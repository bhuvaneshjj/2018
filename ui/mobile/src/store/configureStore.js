import AsyncStorage from 'react-native'; // :TODO: use @react-native-community/async-storage
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import wsAPI from '../ui-common/store/wsAPI';
import { websocketConnected, websocketDisconnected } from '../ui-common/App/connectedActions';
import appReducer from '../App/appReducer';

export const loadState = async () => {
  try {
    const serializedState = await AsyncStorage.getItem('state');
    if (serializedState === null || serializedState === undefined) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while loading app state from the async storage', err);
    return undefined;
  }
};

const saveState = async state => {
  try {
    const serializedState = JSON.stringify(state);
    await AsyncStorage.setItem('state', serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while storing app state to the async storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onBroadcast: ({ broadcast }) => {
      switch (broadcast) {
        /* :TODO:
        case BROADCAST_UCASTNIK:
          store.dispatch(broadcastUcastnik(data));
          break;
        case BROADCAST_STOPKY:
          store.dispatch(broadcastStopky(data));
          break;
        */
        default:
          break;
      }
    },
    onConnect: async () => {
      await store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam) => {
  const initialState = initialStateParam || {};

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(thunk.withExtraArgument(wsClient), wsAPI.withExtraArgument(wsClient))
    )
  );

  store.subscribe(async () => {
    const state = store.getState();
    await saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      serviceURL: state.serviceURL
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
