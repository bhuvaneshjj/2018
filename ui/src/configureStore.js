import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import appReducer from './App/appReducer';
import { websocketConnected, websocketDisconnected } from './App/AppActions';
import { setHighestMezicasId } from './casomeric/Mezicasy/MezicasyActions';

const demoStartujiciState = {
  startujici: [
    { id: 0, cislo: 7, dokonceno: null },
    { id: 1, cislo: 4, dokonceno: null },
    { id: 17, cislo: 16, dokonceno: null },
    { id: 2, cislo: 25, dokonceno: false },
    { id: 3, cislo: 9, dokonceno: null },
    { id: 12, cislo: 15, dokonceno: null },
    { id: 9, cislo: 1, dokonceno: false },
    { id: 29, cislo: 8, dokonceno: null },
    { id: 5, cislo: 59, dokonceno: null },
    { id: 11, cislo: 43, dokonceno: null },
    { id: 18, cislo: 42, dokonceno: false },
    { id: 10, cislo: 33, dokonceno: null },
    { id: 7, cislo: 21, dokonceno: false },
    { id: 13, cislo: 24, dokonceno: null },
    { id: 34, cislo: 27, dokonceno: null },
    { id: 14, cislo: 22, dokonceno: null },
    { id: 58, cislo: 30, dokonceno: null },
    { id: 15, cislo: 23, dokonceno: null },
    { id: 59, cislo: 26, dokonceno: null }
  ]
};

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onConnect: () => {
      store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};
  initialState.startujici = demoStartujiciState.startujici;

  if (initialState.casomeric && initialState.casomeric.mezicasy) {
    let highestId = 0;

    initialState.casomeric.mezicasy.forEach(mezicas => {
      if (mezicas.id > highestId) {
        highestId = mezicas.id;
      }
    });

    setHighestMezicasId(highestId);
  }

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(applyMiddleware(thunk.withExtraArgument(wsClient)))
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      casomeric: { stopky: state.casomeric.stopky, mezicasy: state.casomeric.mezicasy }
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
