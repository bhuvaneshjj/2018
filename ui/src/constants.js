export const TYPY_KATEGORII = ['maraton', 'půlmaraton', 'cyklo', 'koloběžka', 'pěší'];

export const AKTUALNI_ROK = 2018;
export const DEN_ZAVODU = false;

export const STOPKY_ONE_TICK = 10; // milliseconds

export const PRIHLASKY_SAVE_MODAL_TIMEOUT = 2 * 1000; // 2 seconds

export const WEBSOCKET_RECONNECT_INTERVAL = 2 * 1000; // 2 seconds
export const WEBSOCKET_REQUEST_TIMEOUT = 20 * 1000; // 20 seconds
