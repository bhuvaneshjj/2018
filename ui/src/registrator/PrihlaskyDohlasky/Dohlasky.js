import React from 'react';
import PropTypes from 'prop-types';
import PrihlaskyDohlasky from './PrihlaskyDohlasky';

const Dohlasky = ({ match }) => (
  <PrihlaskyDohlasky actionPrefix="DOHLASKY" match={match} name="Dohlášky" reduxName="dohlasky" />
);

Dohlasky.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string.isRequired
  })
};

export default Dohlasky;
