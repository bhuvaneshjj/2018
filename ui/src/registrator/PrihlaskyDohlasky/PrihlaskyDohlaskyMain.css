.PrihlaskyDohlaskyMain__div {
  display: flex;
  flex-direction: column;
}

.PrihlaskyDohlaskyMain__item {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-top: 2ex;
  margin-bottom: 1ex;
}

.PrihlaskyDohlaskyMain__item > * {
  padding-right: 1em;
}

.PrihlaskyDohlaskyMain__icon {
  font-size: xx-large;
}
