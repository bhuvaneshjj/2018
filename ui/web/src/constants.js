export const TYPY_KATEGORII = ['maraton', 'půlmaraton', 'cyklo', 'koloběžka', 'pěší'];

export const AKTUALNI_ROK = 2019;

export const STOPKY_ONE_TICK = 10; // milliseconds

export const TIMESYNC_INITIAL_BURST_DELAY = 1000; // 1 second
export const TIMESYNC_OPERATIONAL_DELAY = 10000; // 10 seconds
export const TIMESYNC_LAST_SAMPLES = 5;

export const PRIHLASKY_SAVE_MODAL_TIMEOUT = 2 * 1000; // 2 seconds

export const WEBSOCKET_RECONNECT_INTERVAL = 2 * 1000; // 2 seconds
export const WEBSOCKET_REQUEST_TIMEOUT = 20 * 1000; // 20 seconds

export const ActionPrefixes = {
  DOHLASENI: 'DOHLASENI',
  DOHLASKY: 'DOHLASKY',
  POHARY_PO_STARTU: 'POHARY_PO_STARTU',
  POHARY_PRED_STARTEM: 'POHARY_PRED_STARTEM',
  PORADI: 'PORADI',
  PRIHLASENI: 'PRIHLASENI',
  PRIHLASKY: 'PRIHLASKY'
};

export const DragTypes = {
  POHAR: 'pohár',
  STARTOVNI_CISLO: 'startovniCislo',
  STARTUJICI_PRIHLASEN: 'startujiciPrihlasen',
  STARTUJICI_ODSTARTOVAN: 'startujiciOdstartovan'
};

export const ReduxNames = {
  dohlaseni: 'dohlaseni',
  dohlasky: 'dohlasky',
  poharyPoStartu: 'poharyPoStartu',
  poharyPredStartem: 'poharyPredStartem',
  poradi: 'poradi',
  prihlaseni: 'prihlaseni',
  prihlasky: 'prihlasky'
};
