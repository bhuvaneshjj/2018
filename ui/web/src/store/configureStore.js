import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { BROADCAST_STOPKY, BROADCAST_UCASTNIK } from 'ui-common/common';
import wsAPI from 'ui-common/store/wsAPI';
import appReducer from '../App/appReducer';
import { websocketConnected, websocketDisconnected } from '../App/AppActions';
import { broadcastStopky } from '../entities/stopky/stopkyActions';
import { broadcastUcastnik } from '../entities/ucastnici/ucastniciActions';
import { timesyncOperation, timesyncStart } from '../Timesync/TimesyncActions';

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onBroadcast: ({ broadcast, data }) => {
      switch (broadcast) {
        case BROADCAST_UCASTNIK:
          store.dispatch(broadcastUcastnik(data));
          break;
        case BROADCAST_STOPKY:
          store.dispatch(broadcastStopky(data));
          break;
        default:
          break;
      }
    },
    onConnect: async () => {
      await store.dispatch(websocketConnected());
      store.dispatch(timesyncOperation());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(thunk.withExtraArgument(wsClient), wsAPI.withExtraArgument(wsClient))
    )
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      }
      // :TODO: casomeric.mezicasy
    });
  });

  // The time synchronization gets started by default.
  store.dispatch(timesyncStart());

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
