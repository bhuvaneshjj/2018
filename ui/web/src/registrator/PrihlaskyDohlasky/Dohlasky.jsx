import React from 'react';
import PropTypes from 'prop-types';
import { ActionPrefixes, ReduxNames } from '../../constants';
import PrihlaskyDohlasky from './PrihlaskyDohlasky';

const Dohlasky = ({ match }) => (
  <PrihlaskyDohlasky
    actionPrefix={ActionPrefixes.DOHLASKY}
    match={match}
    name="Dohlášky"
    reduxName={ReduxNames.dohlasky}
  />
);

Dohlasky.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string.isRequired,
  }).isRequired,
};

export default Dohlasky;
