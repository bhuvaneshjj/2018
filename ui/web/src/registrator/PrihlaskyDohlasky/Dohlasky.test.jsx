import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { ActionPrefixes, ReduxNames } from '../../constants';
import ucastniciTestData, {
  AKTUALNI_DATUM_KONANI
} from '../../entities/ucastnici/ucastniciTestData';
import Dohlasky from './Dohlasky';
import PrihlaskyDohlaskyMain from './PrihlaskyDohlaskyMain';

const actionPrefix = ActionPrefixes.DOHLASKY;
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const reduxName = ReduxNames.dohlasky;

const store = mockStore({
  ...ucastniciTestData,
  fetchingUcastnici: 'done',
  registrator: {
    [reduxName]: {
      form: {
        saved: false,
        saving: false,
        udaje: { narozeni: {} },
        prihlaska: {},
        platby: [],
        ubytovani: {}
      },
      platby: {
        novaPlatbaMinified: true
      },
      startCislo: { showing: false }
    }
  }
});

it('renders /dohlasky route', () => {
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/dohlasky']}>
        <Route path="/dohlasky" component={Dohlasky} />
      </MemoryRouter>
    </Provider>
  );

  expect(wrapper.find(PrihlaskyDohlaskyMain)).toHaveLength(1);
});

it('renders /dohlasky/edit route', () => {
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/dohlasky/edit']}>
        <Route path="/dohlasky" component={Dohlasky} />
      </MemoryRouter>
    </Provider>
  );

  expect(wrapper.find('Connect(PrihlaskyForm)')).toHaveLength(1);
  expect(wrapper.find('Connect(PrihlaskyForm)').props()).toMatchSnapshot();
  expect(store.dispatch).toHaveBeenCalledTimes(1); // fetchUcastnici
});

it('renders /dohlasky/reset route', () => {
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/dohlasky/reset']}>
        <Route path="/dohlasky" component={Dohlasky} />
      </MemoryRouter>
    </Provider>
  );

  expect(wrapper.find('Connect(PrihlaskyForm)')).toHaveLength(1);
  expect(wrapper.find('Connect(PrihlaskyForm)').props()).toMatchSnapshot();
  expect(store.dispatch).toHaveBeenCalledTimes(3); // fetchUcastnici + Form/reset + Platby/reset
  expect(store.dispatch).toHaveBeenCalledWith({
    type: `${actionPrefix}_RESET`,
    datumKonani: AKTUALNI_DATUM_KONANI
  });
  expect(store.dispatch).toHaveBeenCalledWith({
    type: `${actionPrefix}_NOVA_PLATBA_RESET`,
    datumKonani: AKTUALNI_DATUM_KONANI
  });
});

it('renders /dohlasky/:id route', () => {
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/dohlasky/6f09b1fd371dec1e99b7e1c9']}>
        <Route path="/dohlasky" component={Dohlasky} />
      </MemoryRouter>
    </Provider>
  );

  expect(wrapper.find('Connect(PrihlaskyForm)')).toHaveLength(1);
  expect(wrapper.find('Connect(PrihlaskyForm)').props()).toMatchSnapshot();
  expect(store.dispatch).toHaveBeenCalledTimes(2); // fetchUcastnici + load
  expect(store.dispatch).toHaveBeenCalledWith({
    datumKonani: AKTUALNI_DATUM_KONANI,
    id: '6f09b1fd371dec1e99b7e1c9',
    type: `${actionPrefix}_UCASTNIK_LOAD`,
    udaje: {
      jmeno: 'Martina',
      narozeni: { den: 7, mesic: 12, rok: 1963 },
      obec: 'Zlín',
      pohlavi: 'žena',
      prijmeni: 'Sukdoláková',
      stat: 'Česká republika'
    }
  });
});
