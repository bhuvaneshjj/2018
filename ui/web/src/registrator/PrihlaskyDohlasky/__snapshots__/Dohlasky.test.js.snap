// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders /dohlasky/:id route 1`] = `
Object {
  "actionPrefix": "DOHLASKY",
  "loadId": "6f09b1fd371dec1e99b7e1c9",
  "reduxName": "dohlasky",
}
`;

exports[`renders /dohlasky/edit route 1`] = `
Object {
  "actionPrefix": "DOHLASKY",
  "reduxName": "dohlasky",
}
`;

exports[`renders /dohlasky/reset route 1`] = `
Object {
  "actionPrefix": "DOHLASKY",
  "reduxName": "dohlasky",
  "reset": true,
}
`;
